File: leetcode/0019-remove-nth-node-from-end-of-list/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        curr=[]
        while head:
            curr.append(head.val)
            head=head.next
        if len(curr)==0:
            return None
        curr.pop(len(curr)-n)
        if curr==[]:
            return None
        res=ListNode(0)
        pointer=res
        for i in range(0,len(curr)):
            pointer.next=ListNode(curr[i])
            pointer=pointer.next
        return res.next
        """
        :type head: Optional[ListNode]
        :type n: int
        :rtype: Optional[ListNode]
        """
        

File: leetcode/0019-remove-nth-node-from-end-of-list/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if(head->next==NULL){
            return NULL;
        }
        int length=0;
        ListNode* curr=head;
        while(curr!=NULL){
            length++;
            curr=curr->next;
        }
        int pos=length-n-1;
        curr=head;
        
        while(pos>0){
            curr=curr->next;
            pos--;
        }
        if(pos==-1){
            head=head->next;
        }
        else{curr->next=curr->next->next;}
        return head;
    }
};
